psd-web/app/models/user.rb

```diff
@@ -181,9 +185,37 @@ def invitation_expired?
     invited_at <= INVITATION_EXPIRATION_DAYS.days.ago
   end
 
-private
+  def two_factor_authentication_code_expired?
+    return false if !direct_otp_sent_at
+
+    (direct_otp_sent_at + User.direct_otp_valid_for) < Time.current
+  end
+
+  def two_factor_lock_expired?
+    return true if !second_factor_attempts_locked_at
+
+    (second_factor_attempts_locked_at + TWO_FACTOR_LOCK_TIME) < Time.current
+  end
+
+  def lock_two_factor!
+    self.update_column(:second_factor_attempts_locked_at, Time.current)
+  end
+
+  def unlock_two_factor!
+    self.update_columns(second_factor_attempts_locked_at: nil)
```

Comment:
```suggestion
    self.update_column(second_factor_attempts_locked_at: nil)
```
Minor edit for consistency with previous method.

Reply:
@slorek `update_column` takes two arguments it would be  `self.update_column(second_factor_attempts_locked_at,  nil)`